### DISCUSSION TITLE
How about a TUI?

### DISCUSSION THREAD

**Discussion by orhun** [Wuelle/Stormlicht#18] [Ideas]
Hey! Maintainer of [Ratatui](https://ratatui.rs/) here 👋🏼 

I came across your project and it looks awesome! I realized the interface is looking a bit TUI-ish so I wanted to bring up the conversation of using a TUI library (such as `ratatui`) inside the browser. That's an intriguing idea for me and I'm just wondering your opinion on this.

Thanks!

**Comment by simonwuelker** [comment]
Hey, thanks for your nice words!

>  I realized the interface is looking a bit TUI-ish

I'm not entirely sure what you mean by this tbh. I think the idea of a browser-in-the-terminal is interesting (see [lynx](https://lynx.invisible-island.net/) for example), but the rendering seems too limited to me.

Even though the engine is very immature at this point, there are already a number of things that I'm not sure are possible in a terminal, like:
* Images
* Precise alignment of content (to the pixel)
* Overlapping Content

And of course, this list is only going to grow as time goes on (think `:hover` rules, animations, `<canvas>` elements)...

 If someone makes it happen i think that would be very cool; but it is not something that i would consider doing myself. I currently have enough work on my hands trying to make this work on a regular gui :stuck_out_tongue:

**Reply by orhun** [reply]
Yup, one other nice project to look at is [carbonyl](https://github.com/fathyb/carbonyl) I think.

**Reply by simonwuelker** [reply]
Oh wow, that's impressive!
